File: redsheet-portal/src/app/database-models/audit-record.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { AuditRequest, AuditRecordDetail,  } from './'
export class AuditRecord {
    constructor(auditRecord?: AuditRecord )  {
        let properties = [ "auditRecordId", "eventType", "newValue", "oldValue", "recordId", "recordName", "tableName", "auditRequestId", "auditRecordDetails",];
        for (let property of properties)
            if (auditRecord && auditRecord[property])
                this[property] = auditRecord[property];
    }
 
	auditRecordId : number =   0 ;
 
    @required()
    @maxLength(9)
	eventType : string =   undefined;
 
    @required()
	newValue : string =   undefined;
 
	oldValue : string =   undefined;
 
    @range(1,2147483647)
	recordId : number =   undefined;
 
    @required()
	recordName : string =   undefined;
 
    @required()
    @maxLength(50)
	tableName : string =   undefined;
 
	auditRequestId : number =   undefined;
	auditRequest : AuditRequest  ;
	@nested(AuditRecordDetail)
	auditRecordDetails: AuditRecordDetail[];



}

File: RedSheet.DbEntities/Models/AuditRecord.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("AuditRecords", Schema = "dbo")]
	[Serializable]
	public partial class AuditRecord  
	{
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int AuditRecordId { get; set; } 
 
	[Required] 
	[MaxLength(9)]
	public string EventType { get; set; } 
 
	[Required]
	public string NewValue { get; set; } 

	public string OldValue { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int RecordId { get; set; } 
 
	[Required]
	public string RecordName { get; set; } 
 
	[Required] 
	[MaxLength(50)]
	public string TableName { get; set; } 
 
	[RelationshipTableAttribue("AuditRequests","dbo")]
	//Relationships
	public Nullable<int> AuditRequestId { get; set; } 
 
	[ForeignKey("AuditRequestId")]
	public AuditRequest AuditRequest { get; set; } 
	[InverseProperty("AuditRecord")]
	public ICollection<AuditRecordDetail> AuditRecordDetails { get; set; } 


	public AuditRecord()
	{
	this.AuditRecordDetails = new HashSet<AuditRecordDetail>();

    }

    }
}

